x-pack/legacy/plugins/uptime/public/components/functional/ping_list/ping_list.tsx

```diff
@@ -20,109 +20,135 @@ import {
 } from '@elastic/eui';
 import { i18n } from '@kbn/i18n';
 import { FormattedMessage } from '@kbn/i18n/react';
-import { get } from 'lodash';
 import moment from 'moment';
-import React, { Fragment, useState } from 'react';
+import React, { useState, useEffect } from 'react';
 import styled from 'styled-components';
-import { CriteriaWithPagination } from '@elastic/eui/src/components/basic_table/basic_table';
-import { Ping, PingResults } from '../../../../common/graphql/types';
+import { Ping, GetPingsParams } from '../../../../common/runtime_types';
 import { convertMicrosecondsToMilliseconds as microsToMillis } from '../../../lib/helper';
-import { UptimeGraphQLQueryProps, withUptimeGraphQL } from '../../higher_order';
-import { pingsQuery } from '../../../queries';
 import { LocationName } from './location_name';
 import { Pagination } from './../monitor_list';
 import { PingListExpandedRowComponent } from './expanded_row';
+import { PingListProps } from '../../connected/pings';
 
-interface PingListQueryResult {
-  allPings?: PingResults;
-}
-
-interface PingListProps {
-  onSelectedStatusChange: (status: string | undefined) => void;
-  onSelectedLocationChange: (location: any) => void;
-  onPageCountChange: (itemCount: number) => void;
-  onPageIndexChange: (index: number) => void;
-  pageSize: number;
-  pageIndex: number;
-  selectedOption: string;
-  selectedLocation: string | undefined;
-}
-
-type Props = UptimeGraphQLQueryProps<PingListQueryResult> & PingListProps;
 interface ExpandedRowMap {
   [key: string]: JSX.Element;
 }
 
-export const AllLocationOption = { text: 'All', value: '' };
+export const AllLocationOption = {
+  'data-test-subj': 'xpack.uptime.pingList.locationOptions.all',
+  text: 'All',
+  value: '',
+};
 
 export const toggleDetails = (
   ping: Ping,
   itemIdToExpandedRowMap: ExpandedRowMap,
   setItemIdToExpandedRowMap: (update: ExpandedRowMap) => any
 ) => {
   // If the user has clicked on the expanded map, close all expanded rows.
-  if (itemIdToExpandedRowMap[ping.id]) {
+  if (itemIdToExpandedRowMap[ping['@timestamp']]) {
```

Comment:
is id not unique?

Reply:
I think not for the pings list, as it's a history of many pings for the same monitor, all those documents are going to have the same `id` field, but none should have the same `@timestamp`.

Reply:
Had some changes impacting this code in https://github.com/elastic/kibana/pull/59392/commits/433bb7475a367ec6b573e7fadc94ce578733d468.